BasicJSONTableView/APIService.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  APIService.swift
+//  BasicJSONTableView
+//
+//  Created by Nick Nguyen on 10/8/20.
+//  Copyright © 2020 Declan McKenna. All rights reserved.
+//
+
+import Foundation
+
+class APIService {
+
+    let session: URLSession
+    let jsonDecoder = JSONDecoder()
+    var cards: Cards?
+
+    lazy var urlRequest: URLRequest = {
+        let url = URL(string: "https://deckofcardsapi.com/api/deck/new/draw/?count=12#")!
+        return URLRequest(url: url)
+    }()
+
+    init(session: URLSession = URLSession.shared) {
+        self.session = session
+    }
+
+    func getCardsFromServer(completion: @escaping (Result<[Card],Error>) -> Void) {
+        session.dataTask(with: self.urlRequest) { data, response, error in
+
+            guard let data = data else { fatalError("Missing data") }
```

Comment:
I like using fatal errors a lot more than force unwrapping when we've reached a state that out app can't sustain.
I think a throwing an error or given the time limit just printing a log might be better here as otherwise we will crash when the app is offline.